### ISSUE TITLE
Masks not predicted, only bboxes TF 2.5

### ISSUE THREAD

**Issue by valavanisleonidas** [matterport/Mask_RCNN#2809]
Hello everyone,

I am using Maskrcnn with TF 2.5 from this [forked repo](https://github.com/sabderra/Mask_RCNN) and training with my own dataset. I have 1200 images for train and 100 for validation. I have only one class, which is corrosion and the objects might be large or small. On my first training I saw that while the bboxes are good enough, the mask is overfitting and all the values are < 0.001 so there is no mask shown in the images. Btw I am using coco pretrained model and training only the heads.

The loss for the mask quickly goes to 0.00x which I guess means that it learns to predict everything black, thus overfitting

I experimented using resnet50 since the images are not many and maybe resnet101 is too complex for the data but no luck yet

Any suggestions on how to get good masks ??

Appreciate the help

**Comment by Kamlesh364** [comment]
> Hello everyone,
> 
> I am using Maskrcnn with TF 2.5 from this [forked repo](https://github.com/sabderra/Mask_RCNN) and training with my own dataset. I have 1200 images for train and 100 for validation. I have only one class, which is corrosion and the objects might be large or small. On my first training I saw that while the bboxes are good enough, the mask is overfitting and all the values are < 0.001 so there is no mask shown in the images. Btw I am using coco pretrained model and training only the heads.
> 
> The loss for the mask quickly goes to 0.00x which I guess means that it learns to predict everything black, thus overfitting
> 
> I experimented using resnet50 since the images are not many and maybe resnet101 is too complex for the data but no luck yet
> 
> Any suggestions on how to get good masks ??
> 
> Appreciate the help

Hey there!
[this](https://github.com/Kamlesh364/Mask-RCNN-TF2.7.0-keras2.8.0) development may help you
